src/datatrove/pipeline/tokens/tokenizer.py

```diff
@@ -150,7 +150,7 @@ def write(self, tokens: list[int], loss_values: np.ndarray | None):
             loss_values (np.ndarray | None): optional loss values to write
         """
         # get the bytes for uint16 (H)
-        self.write_bytes(struct.pack("<%sH" % len(tokens), *tokens))
```

Comment:
This has to be made optional (with the default being uint16 as before). There are also other changes required (size computations using "2" and the tokenizer merger, for example).